src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/external/UserSearchController.kt

```diff
@@ -24,6 +30,56 @@ class UserSearchController(
   private val userSearchService: UserSearchService
 ) {
 
+  @GetMapping("/search")
+  @Operation(
+    summary = "Search for an external user."
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "200",
+        description = "OK"
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      )
+    ]
+  )
+  @PreAuthorize(
+    "hasAnyRole('ROLE_MAINTAIN_OAUTH_USERS', 'ROLE_AUTH_GROUP_MANAGER')"
+  )
+  suspend fun searchForUser(
+    @Parameter(
+      description = "The username, email or name of the user.",
+      example = "j smith"
+    ) @RequestParam(required = false)
+    name: String?,
+    @Parameter(description = "The role codes of the user.") @RequestParam(required = false)
+    roles: List<String>?,
+    @Parameter(description = "The group codes of the user.") @RequestParam(required = false)
+    groups: List<String>?,
+    @Parameter(description = "Limit to active / inactive / show all users.") @RequestParam(
+      required = false,
+      defaultValue = "ALL"
+    )
+    status: Status,
+    @PageableDefault(sort = ["Person.lastName", "Person.firstName"], direction = Sort.Direction.ASC) pageable: Pageable
+  ): Page<UserDto> =
```

Comment:
I don't fully understand the default sort - there is no "Person" object?

Reply:
Was just about to say that this was copied from Auth and there is no Person object there either, but just realised that is not true! The ORDER By is in fact hard coded in external-users (pretty sure we discussed this at the time). That being the case I'm not sure whether either of the APIs should be giving the impression that it can be changed?